### ISSUE TITLE
IGNORE_UNKNOWN on simple class

### ISSUE THREAD

**Issue by kanbara** [FasterXML/jackson-dataformat-csv#121]
I'm having a ton of issues with dataformat-csv 2.7.2 on a really simple class. I originally this had an abstract class as a mixin which worked, but I realized that the data I needed was on an object in the object, which was giving me grief, so I made a simple class to attempt to lighten the complexity of my serialization attempts.

When I run this, I get IGNORE_UNKNOWN as returned by the Response object. I was getting similar things with the full object too-- if all the properties are accounted for, I get a JsonMappingException which somehow doesn't get thrown, but again added to the Response. Not really sure what's going on.

It appears that on the `String csv =  writer.writeValueAsString(items);` it encounters an error and forces the mule service to return without catching the exception, so I can't even tell what the actual problem is.

```java
public class TileOrderCsvFormat {

  private int imagetakeTileCatId;

  public TileOrderCsvFormat(TileOrder to) {
    this.imagetakeTileCatId = to.getImagetakeTileCatId();
  }

  public int getImagetakeTileCatId() {
    return imagetakeTileCatId;
  }
  public void setImagetakeTileCatId(int imagetakeTileCatId) {
    this.imagetakeTileCatId = imagetakeTileCatId;
  }
}
```

The jackson code:

```java
try {
      
      List<TileOrderCsvFormat> itemsForCsvFormat = new ArrayList<TileOrderCsvFormat>();
      
      for(TileOrder to : items) {
        itemsForCsvFormat.add(new TileOrderCsvFormat(to));
      }
            
      CsvMapper mapper = new CsvMapper();
      CsvSchema schema = mapper.schemaFor(TileOrderCsvFormat.class).withHeader();
      ObjectWriter writer = mapper.writer(schema);
      String csv =  writer.writeValueAsString(items);
      Response res = Response.status(Status.OK).entity(csv).build();
      return res;
      
    } catch (Exception ex) {
 
      log.info(errorMessage.toString(), ex);
      String csvResponse = RestResponseUtils.constructErrorBaseRestResponseJson(errorMessage.toString(), "", serviceName);
      return Response.status(Status.INTERNAL_SERVER_ERROR).entity(csvResponse).build();
    }
    
  }
```

**Comment by cowtowncoder** [comment]
Ok. First things first: which version of Jackson is this with?

Also: what exactly is the problem? `IGNORE_UNKNOWN` does not say much; there is a `SerializationFeature` with that name, but how is that related?
If I remove container-specific parts (`Response`), does above fail with specific exception?

**Comment by kanbara** [comment]
Jackson and dataformat 2.7.2.

I tried debugging and backtracing but I didn't really get anywhere. The `IGNORE_UNKNOWN` or `JsonMappingException` are what my Response spits out. The code appears to return from the method without calling the Response line, or catching + logging the exception, as I put some log lines in that wouldn't fire. I tested the above with a library as well (so no response, just returning a string of csv) and it did the same.

I use Jackson with POJO->JSON just fine, but I can't seem to get the dataformat-csv to behave as well. Not sure if using mule has anything to do with it, as it brings in some of its Jackson libraries-- some of which are 1.9.11.

**Comment by cowtowncoder** [comment]
Ok. Version discrepancy between 1.x and 2.x should not matter: components live in different Java and Maven packages, so should co-exist peacefully. 1.x is typically brought in as Hadoop or Avro dependency.

I think it would be great if this could be reproduced with a test that is not contained within JAX-RS (etc) `Response` object. I do not see anything unusual about object itself, so that should be fine.

**Comment by kanbara** [comment]
I added an `ExceptionMapper<Throwable>` implementation, which gave me the exception.

```
2016-03-02T10:51:01,957 ERROR:77 - exception: 
java.lang.NoSuchFieldError: IGNORE_UNKNOWN
	at com.fasterxml.jackson.dataformat.csv.CsvGenerator._writeFieldName(CsvGenerator.java:368) ~[jackson-dataformat-csv-2.7.2.jar:2.7.2]
	at com.fasterxml.jackson.dataformat.csv.CsvGenerator.writeFieldName(CsvGenerator.java:340) ~[jackson-dataformat-csv-2.7.2.jar:2.7.2]
	at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:503) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:639) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:152) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:100) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:21) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ser.std.AsArraySerializerBase.serialize(AsArraySerializerBase.java:183) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:114) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ObjectWriter._configAndWriteValue(ObjectWriter.java:800) ~[jackson-databind-2.4.3.jar:2.4.3]
	at com.fasterxml.jackson.databind.ObjectWriter.writeValueAsString(ObjectWriter.java:676) ~[jackson-databind-2.4.3.jar:2.4.3]

```

**Comment by cowtowncoder** [comment]
Hmmh. That certainly sounds weird... but it strongly sounds like you might have duplicate jars in the classpath, and specifically old(er) version of `jackson-core`. Although this has to be quite old, as `JsonGenerator.Feature.IGNORE_UNKNOWN` was added in 2.5, so probably 2.4.4 pulled in as transitive dependency.

**Comment by kanbara** [comment]
Yeah it looks like that's the case unfortunately. I fixed it by using 2.4.3
as Mule is stubborn when overloading the class path

**Comment by cowtowncoder** [comment]
@kanbara Ok good that you are making progress at least.
